ocaml/xapi/quicktest_cbt.ml

```diff
@@ -0,0 +1,112 @@
+(*
+ * Copyright (C) 2017 Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Client
+open Quicktest_common
+exception Test_assertion_failed of string
+
+let start session_id =
+  let open Client in
+
+  (* Helper for test failure due to unexpected error *)
+  let report_failure error test_name test =
+    failed test (Printf.sprintf "%s failed: %s" test_name
+                   (ExnHelper.string_of_exn error)) in
+
+  (* Define own exception to return error message to the test try/with and fail within that,
+   * as calling 'failed test' in a function called within the test leads to Hashtbl problems *)
+  let test_assert ~test op ~msg =
+    if not op then raise (Test_assertion_failed msg) in
+
+  let name_description = "VDI for CBT quicktest" in
+  let make_vdi_from sR = (* SR has VDI.create as allowed *)
+    VDI.create
+      ~sR
+      ~session_id
+      ~rpc:!rpc
+      ~name_label:"qt-cbt"
+      ~name_description
+      ~_type:`user
+      ~sharable:false
+      ~read_only:false
+      ~virtual_size:(2L ** mib)
```

Comment:
Just one last thing: I've just realized that it's safer to create a 4MiB VDI, because LVM SRs don't support less than 4MB VDIs